docs/debugging-s2i.md

```diff
@@ -0,0 +1,116 @@
+
+Troubleshooting
+=================
+
+This document contains some tips and suggestions for troubleshooting the use of source-to-image ("s2i") to inject source into Docker images.
+
+Can't Find The `sti` Executable
+--------------
+
+Whether by  running `make build`,  running `hack/build-release.sh`, or downloading one of our official releases,  look under the `_output/local/go/bin` directory.
+
+(GGM NOTE:  stems from https://github.com/openshift/source-to-image/issues/74 )
+
+Image Mechanics
+--------------
+
+By default, s2i will use the local builder image if available on the host you are running the `sti` command from, instead of pulling it from the remote docker registry.  If that image becomes stale and out of date, either use the `--forcePull=true` option when running the `sti` command,
+or manually pull the correct image to the host from where you are running the `sti` command.
+
+Permissions Needed During the S2I Process
+--------------
+
+The s2i process will levarage the `/tmp` directory at various stages, such as when cloning the source repository, when augmenting the Dockerfile for layered builds, during various `tar` operations that are performed, and when executing of the scripts you have provided.  Invalid permissions will result in a message like this:
+
+     E0720 21:07:17.145257 04202 main.go:328] An error occurred: creating temporary directory  failed
+
+Hence, the user ID you execute the `sti` command with must have the appropriate permission to read, create, and remove files under `/tmp`.
+
+(GGM NOTE:  came from a github closed issue I thought I saw on Friday which I can't find currently (or maybe it was an email thread) )
+
+Passing In Environment Variables
+--------------
+
+If you are passing in environment variables with the -e option from the command line, and those environment variables have values with commas in them, Go command line processing will break.
+
+       $ ~/sandbox/ruby sti build --loglevel=5 -e "MAVEN_ARGS='-P some-repo,some-other-repo'" file:///home/mfojtik/sandbox/ruby openshift/ruby-20-centos7 foo
+       	 I0519 14:20:51.410089 17228 util.go:34] Using mi@mifo.sk credentials for pulling openshift/ruby-20-centos7
+     	 I0519 14:20:51.410222 17228 main.go:239] An error occurred: malformed env string: some-other-repo'
+
+To deal with this behavior, both the -E option and the `.sit/environment` file allow for the values of environment variable to contain commas.
```

Comment:
More info, or a sample or at least link to doc explaining how to actually use those.